### PR TITLE
refactor: simplify data mapping for reports

### PR DESCRIPTION
**Pull request by TheMaskedTurtle**

### CODE REVIEW

**Review by dbraquart — CHANGES_REQUESTED** [review]
code : a few remarks
tests: 1 test case ko
- in incremental built, severity is not set at the node level

File: src/utils/report-tree.mapper.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { Report, ReportTree } from '../types/report.type';
+import { getHighestSeverity } from './report-severity.utils';
+import { GLOBAL_NODE_TASK_KEY, REPORT_TYPE } from '../constants/report.constant';
+
+export function mapReportsTree(report: Report, reportType?: string): ReportTree {
+    const severityList = report.severities || report.subReports.flatMap((subReport) => subReport.severities);
```
[the remark this answers is not in the document]
Comment (TheMaskedTurtle) [reply src/utils/report-tree.mapper.ts:13]: I still need this one, only for one case, to evaluate severities for "Logs" container

File: src/components/report-viewer/report-viewer.jsx [hunk 2]
```diff
@@ -5,121 +5,71 @@
...
-export default function ReportViewer({
-    jsonReportTree,
-    subReportPromise,
-    nodeReportPromise,
-    globalReportPromise = undefined,
-    maxSubReports = MAX_SUB_REPORTS,
-}) {
-    const [selectedNode, setSelectedNode] = useState(null);
-    const [expandedNodes, setExpandedNodes] = useState([]);
+export default function ReportViewer({ report, reportType, maxSubReports = MAX_SUB_REPORTS }) {
```
[the remark this answers is not in the document]
Comment (TheMaskedTurtle) [reply src/components/report-viewer/report-viewer.jsx:29]: done

File: src/components/results/common/computation-report-viewer.tsx [hunk 3]
```diff
@@ -5,88 +5,40 @@
...
-            // an array with a single reporter is expected (corresponding to the current Node)
-            let singleReport: any = Array.isArray(reportData) && reportData.length === 1 ? reportData[0] : undefined;
-            if (nodeName && singleReport) {
-                singleReport.title = nodeName;
-            }
-            return singleReport;
-        },
-        [currentNode?.data.label]
-    );
+    const { snackError } = useSnackMessage();
```
[the remark this answers is not in the document]
Comment (TheMaskedTurtle) [reply src/components/results/common/computation-report-viewer.tsx:26]: weird this was not linted for me :thinking:

File: src/hooks/use-report-fetcher.tsx [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        return {
+            message: GLOBAL_NODE_TASK_KEY,
+            id: GLOBAL_NODE_TASK_KEY,
+            subReports: report.map((r) => setNodeName(r, nodesNames)),
+        } as Report;
+    }
+}
+
+function setNodeName(report: Report, nodesNames: Map<string, string>) {
+    if (report.message !== 'Root') {
```
[the remark this answers is not in the document]
Comment (TheMaskedTurtle) [reply src/hooks/use-report-fetcher.tsx:38]: created a new constant file as it is more related to nodes

File: src/utils/report-tree.mapper.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (dbraquart) [reply src/utils/report-tree.mapper.ts:13]: ok!

File: src/components/results/common/computation-report-viewer.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (dbraquart) [reply src/components/results/common/computation-report-viewer.tsx]: me neither, I just noticed

**Pull request merged**